File: dmis-api/scripts/extract_model_refs.py
import os
import json
import re
from pathlib import Path

dir_path = Path(__file__).resolve().parent.parent / 'models'
results = []
for file in sorted(dir_path.glob('*.js')):
    text = file.read_text(encoding='utf-8')
    model_match = re.search(r"export default mongoose\.model\(['\"](.*?)['\"]", text)
    model_name = model_match.group(1) if model_match else file.stem
    collection_match = re.search(r"collection\s*:\s*['\"](.*?)['\"]", text)
    collection = collection_match.group(1) if collection_match else None
    refs = sorted(set(re.findall(r"ref\s*:\s*['\"](.*?)['\"]", text)))
    idx_count = len(re.findall(r"\.index\(|index\(", text))
    results.append({
        'file': file.name,
        'model': model_name,
        'collection': collection,
        'refs': refs,
        'indexCount': idx_count,
    })
print(json.dumps(results, indent=2))

File: dmis-api/scripts/extract_model_refs.js
import fs from 'fs';
import path from 'path';
const dir = path.resolve('models');
const files = fs.readdirSync(dir).filter(f => f.endsWith('.js')).sort();
const results = [];
for (const file of files) {
  const text = fs.readFileSync(path.join(dir, file), 'utf8');
  const modelMatch = text.match(/export default mongoose\.model\(['\"](.*?)['\"]/);
  const model_name = modelMatch ? modelMatch[1] : file.replace('.js', '');
  const collectionMatch = text.match(/collection\s*:\s*['\"](.*?)['\"]/);
  const collection = collectionMatch ? collectionMatch[1] : null;
  const refs = Array.from(new Set([...text.matchAll(/ref\s*:\s*['\"](.*?)['\"]/g)].map(m => m[1])));
  const idx_count = [...text.matchAll(/\.index\(|index\(/g)].length;
  results.push({ file, model: model_name, collection, refs, indexCount: idx_count });
}
console.log(JSON.stringify(results, null, 2));
